render_files.py

```diff
@@ -77,6 +84,40 @@ def debug_print_upload_to_s3(install_dir):
             print("Uploading file %s as object %s" % (file_path, s3_file_name))
 
 
+def install_config_manipulation(before_installer, config_dir):
+    if before_installer:
+        shutil.copyfile(os.path.join(config_dir, INSTALL_CONFIG), os.path.join(config_dir, INSTALL_CONFIG_BACKUP))
+    else:
+        shutil.move(os.path.join(config_dir, INSTALL_CONFIG_BACKUP), os.path.join(config_dir, INSTALL_CONFIG))
+
+
+def generate_installation_files(work_dir, config_dir):
+    # command = "OPENSHIFT_INSTALL_INVOKER=\"assisted-installer\" %s/openshift-baremetal-install create ignition-configs --dir %s" \
+    #        % (work_dir, config_dir)
+    command = "OPENSHIFT_INSTALL_INVOKER=\"assisted-installer\" %s/openshift-install create " \
+              "ignition-configs --dir %s" % (work_dir, config_dir)
+    try:
+        subprocess.check_output(command, shell=True, stderr=sys.stdout)
+    except Exception as ex:
+        raise Exception('Failed to generate files, exception: {}'.format(ex))
+
+    install_config_manipulation(config_dir=config_dir, before_installer=False)
+
+
+def prepare_install_config(config_dir, install_config):
+    if install_config:
+        subprocess.check_output(["mkdir", "-p", config_dir])
+        with open(os.path.join(config_dir, INSTALL_CONFIG), 'w+') as file_obj:
+            file_obj.write(install_config)
+    if not os.path.isdir(config_dir):
+        raise Exception('installer directory is not mounted')
+
+    if not os.path.isfile(os.path.join(config_dir, INSTALL_CONFIG)):
+        raise Exception("install config file not located in installer dir")
+
+    install_config_manipulation(config_dir=config_dir, before_installer=True)
```

Comment:
All these if seems irrelevant :wink: 
Consider this:
```suggestion
def prepare_install_config(config_dir, install_config):
    if not install_config:
            raise Exception("install config file not located in installer dir")
    subprocess.check_output(["mkdir", "-p", config_dir])
    install_config_file = os.path.join(config_dir, INSTALL_CONFIG)
    log.info("writing install config to file")
    with open(install_config_file, 'w+') as file_obj:
         file_obj.write(install_config)
    install_config_manipulation(config_dir=config_dir, before_installer=True)
```

Reply:
right , just didn't wanted to change code :) just to move it. But yes your right

Reply:
not sure suggested change matches current logic